x-pack/plugins/security_solution/public/common/store/global_url_param/reducer.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { reducerWithInitialState } from 'typescript-fsa-reducers';
+import { registerUrlParam, updateUrlParam, deregisterUrlParam } from './actions';
+
+export type GlobalUrlParam = Record<string, string | null>;
+
+export const initialGlobalUrlParam: GlobalUrlParam = {};
+
+export const globalUrlParamReducer = reducerWithInitialState(initialGlobalUrlParam)
+  .case(registerUrlParam, (state, { key, initialValue }) => {
+    // It doesn't allow the query param to be used twice
+    if (state[key] !== undefined) {
+      throw new Error(`Url param string '${key}' is already being used.`);
+    }
```

Comment:
That is a valid concern. It happens when you navigate from security to observability and come back. Because the store is preserved, but the components are re-mounted. I implemented `deregisterUrlParam` for that scenario. Now that I am transforming `registerUrlParam` into a hook, I will make sure that `deregisterUrlParam` is called when the component unmounts.

Do you think that a `console.warning` would be a better fit? Throwing an Erro might be too dramatic 😅